pkg/controller/scheduler.go

```diff
@@ -616,11 +616,20 @@ func (c *Controller) runAnalysis(canary *flaggerv1.Canary) bool {
 	return true
 }
 
-func (c *Controller) shouldSkipAnalysis(canary *flaggerv1.Canary, canaryController canary.Controller, meshRouter router.Interface) bool {
+func (c *Controller) shouldSkipAnalysis(canary *flaggerv1.Canary, canaryController canary.Controller, meshRouter router.Interface, err error, retriable bool) bool {
 	if !canary.SkipAnalysis() {
 		return false
 	}
 
+	// regardless if analysis is being skipped, rollback if canary failed to progress
```

Comment:
without this block, flagger immediately tries to promote the new version which at this point has 0 healthy pods.